### PR TITLE
Tw 407 file infor

### PR DESCRIPTION
**Pull request by hoangdat**
### Issue:
- #373: remove other picker type in upload avatar
- and support file info in upload content

### PRs:
- https://github.com/linagora/linagora-design-flutter/pull/20
- here: 

### Demo: 
- 

https://github.com/linagora/twake-on-matrix/assets/6462404/15ef8d2d-40fb-4ae7-8b11-3763903acc8c

### CODE REVIEW

File: lib/domain/usecase/room/upload_content_interactor.dart [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+import 'package:fluffychat/data/network/upload_file/upload_file_api.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
 import 'package:fluffychat/domain/app_state/room/upload_content_state.dart';
 import 'package:fluffychat/domain/exception/room/can_not_upload_content_exception.dart';
-import 'package:fluffychat/presentation/extensions/asset_entity_extension.dart';
+import 'package:fluffychat/presentation/model/file/file_asset_entity.dart';
 import 'package:matrix/matrix.dart';
 import 'package:wechat_camera_picker/wechat_camera_picker.dart';
 
 class UploadContentInteractor {
```
Comment (sherlockvn) [inline lib/domain/usecase/room/upload_content_interactor.dart:12]: should rename the interactor and filename also

File: lib/pages/chat/chat_input_row.dart [hunk 2]
```diff
@@ -226,6 +216,27 @@ class ChatInputRow extends StatelessWidget {
       ),
     );
   }
+
+  void _showMediaPicker(BuildContext context) {
```
Comment (drminh2807) [inline lib/pages/chat/chat_input_row.dart:220]: can we move it to controller ?

File: lib/pages/chat_draft/draft_chat_view.dart [hunk 3]
```diff
@@ -261,6 +251,23 @@ class DraftChatView extends StatelessWidget {
       ),
     );
   }
+
+  void _showMediaPicker(BuildContext context) {
```
Comment (drminh2807) [inline lib/pages/chat_draft/draft_chat_view.dart:255]: same

File: lib/presentation/mixins/common_media_picker_mixin.dart [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+typedef OnSendPhotosTap = void Function()?;
+
+mixin CommonMediaPickerMixin {
+  Future<PermissionStatus>? getCurrentPhotoPermission() {
+    return PermissionHandlerService().requestPermissionForPhotoActions();
+  }
+
+  Future<PermissionStatus>? getCurrentCameraPermission() {
+    return PermissionHandlerService().requestPermissionForCameraActions();
```
Comment (drminh2807) [inline lib/presentation/mixins/common_media_picker_mixin.dart:18]: can we use DI or create instance in Mixin ?

File: lib/presentation/mixins/common_media_picker_mixin.dart [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    Navigator.pop(context);
+    return await CameraPicker.pickFromCamera(
+      context,
+      pickerConfig: onlyImage
+          ? const CameraPickerConfig()
+          : const CameraPickerConfig(
+              enableRecording: true,
+              enableAudio: true,
+            ),
+      // ignore: deprecated_member_use
```
Comment (drminh2807) [inline lib/presentation/mixins/common_media_picker_mixin.dart:65]: window.locale is deprecated
can you investigate the replacement?

File: lib/presentation/mixins/media_picker_mixin.dart [hunk 6]
```diff
@@ -1,92 +1,71 @@
...
         room: room,
-        onItemAction: onItemAction,
-      ).whenComplete(() => removeAllImageSelected());
+        onPickerTypeTap: onPickerTypeTap,
+      );
     }
   }
 
-  Future<void> showImagesPickerBottomSheet({
+  Future<void> showMediasPickerBottomSheet({
```
Comment (drminh2807) [inline lib/presentation/mixins/media_picker_mixin.dart:53]:
```suggestion
  Future<void> showMediaPickerBottomSheet({
```
medias is unknown word

File: lib/presentation/mixins/common_media_picker_mixin.dart (hunk 4 above)
Comment (hoangdat) [reply lib/presentation/mixins/common_media_picker_mixin.dart:18]: what is the DI? Try to minimize the complex of mixin, I think.

Comment (drminh2807) [reply lib/presentation/mixins/common_media_picker_mixin.dart:18]: Dependency Injection
PermissionHandlerService maybe have different implement on Web , Desktop

**Review by sherlockvn — APPROVED** [review]

**Review by drminh2807 — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**